crates/cxx-qt-gen/src/generator/cpp/property/mod.rs

```diff
@@ -60,25 +85,49 @@ pub fn generate_cpp_properties(
 mod tests {
     use super::*;
 
+    use crate::parser::property::QPropertyFlags;
+
     use crate::generator::naming::qobject::tests::create_qobjectname;
     use crate::CppFragment;
     use indoc::indoc;
     use pretty_assertions::assert_str_eq;
     use quote::format_ident;
-    use syn::parse_quote;
+    use syn::{parse_quote, ItemStruct};
+
+    #[test]
+    fn test_optional_write() {
+        let mut input: ItemStruct = parse_quote! {
+            #[qproperty(i32, num, read, write, notify)]
+            struct MyStruct;
+        };
+        let property = ParsedQProperty::parse(input.attrs.remove(0)).unwrap();
+
+        let properties = vec![property];
+
+        let qobject_idents = create_qobjectname();
+
+        let type_names = TypeNames::mock();
+        let _generated =
```

Comment:
ideally you want to be doing the assert_eq and assert_str_eq on the results here to check the right number and content of the things. You can likely copy and maybe even share some from below depending on the property names etc.

Reply:
done